src/clients/MultiCallerClient.ts

```diff
@@ -513,3 +514,138 @@ export class MultiCallerClient {
     }
   }
 }
+
+export class TryMulticallClient extends MultiCallerClient {
+  override _buildMultiCallBundle(transactions: AugmentedTransaction[]): AugmentedTransaction {
+    const txn = super._buildMultiCallBundle(transactions);
+    txn.method = "tryMulticall";
+    return txn;
+  }
+
+  protected override async _executeTxnQueue(
+    chainId: number,
+    txns: AugmentedTransaction[] = [],
+    _valueTxns: AugmentedTransaction[] = [],
+    simulate = false
+  ): Promise<TransactionResponse[]> {
+    // Lint Appeasement
+    _valueTxns;
+    const nTxns = txns.length;
+    if (nTxns === 0) {
+      return [];
+    }
+
+    const networkName = getNetworkName(chainId);
+    this.logger.debug({
+      at: "TryMulticallClient#executeTxnQueue",
+      message: `${simulate ? "Simulating" : "Executing"} ${nTxns} transaction(s) on ${networkName}.`,
+    });
+
+    const buildRawTransaction = (contract: Contract, data: string) => {
+      return {
+        contract,
+        data,
+      } as RawTransaction;
+    };
+
+    const txnRequestsToSubmit: AugmentedTransaction[] = [];
+    const txnCalldataToRebuild: RawTransaction[] = [];
+
+    // The goal is to simulate the transactions as a batch and pick out those which succeed.
+    const bundledTxns = await this.buildMultiCallBundles(txns, this.chunkSize[chainId]);
+    const bundledSimResults = await this.txnClient.simulate(bundledTxns);
+
+    bundledSimResults.forEach(({ succeed, transaction, reason, data }, idx) => {
+      // TryMulticall _should_ always succeed, but either way, we need the return data to be defined so that we can properly
+      // filter out the transactions which failed.
+      if (!succeed || !isDefined(data?.length)) {
+        return;
+      }
+      // Address the case where we just call fillV3Relay(), and therefore the data field is empty.
+      if (succeed && transaction.method !== "tryMulticall") {
+        txnRequestsToSubmit.push(transaction);
+        return;
+      }
+      // If we make it here, then tryMulticall was simulated and there is a defined return data.
+      const succeededTxnCalldata: string[] = [];
+      // We also need to assert that the number of calls which returned data matches the number of calls made in the transaction.
+      assert(transaction.args.length === data.length);
+      // Go through each transaction result of the txns batched in tryMulticall.
+      // If a transaction succeeded, we take note by adding it to succeededTxnCalldata.
+      data.forEach(({ success }, idx) => {
+        if (success) {
+          succeededTxnCalldata.push(transaction.args[idx]);
+        }
+      });
```

Comment:
Could something like this work?

```suggestion
      // If we make it here, then tryMulticall was simulated and there is a defined return data.
      // Verify that the number of calls which returned data matches the number of calls made in the transaction.
      assert(transaction.args.length === data.length);

      // Filter the calldata array by whether it succeeded in tryMulticall().
      const succeededTxnCalldata = transaction.args.filter((callData, idx) => data[idx].success);
```